File: backend/app/schemas/note.py
"""
笔记 Pydantic 模式
"""
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel, ConfigDict


class NoteBase(BaseModel):
    """笔记基础字段"""
    title: str
    content: Optional[str] = None
    note_type: str = "note"
    status: str = "active"


class NoteCreate(NoteBase):
    """创建笔记"""
    pass


class NoteUpdate(BaseModel):
    """更新笔记"""
    title: Optional[str] = None
    content: Optional[str] = None
    note_type: Optional[str] = None
    status: Optional[str] = None


class NoteResponse(NoteBase):
    """笔记响应"""
    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: int
    related_chapter_ids: Optional[List[int]] = None
    sort_order: int
    created_at: datetime
    updated_at: Optional[datetime] = None


class MiaojiParseRequest(BaseModel):
    """妙记解析请求"""
    content: str


class MiaojiParseResult(BaseModel):
    """妙记解析结果"""
    characters: List[dict] = []
    worldbuilding: List[dict] = []
    outline: List[dict] = []
    events: List[dict] = []
    summary: str = ""

File: frontend/src/api/note.ts
/**
 * 笔记/妙记 API
 */
import request from './request'

export interface Note {
  id: number
  project_id: number
  title: string
  content: string | null
  note_type: string
  status: string
  related_chapter_ids: number[] | null
  sort_order: number
  created_at: string
  updated_at: string | null
}

export interface CreateNoteData {
  title: string
  content?: string
  note_type?: string
  status?: string
}

export interface UpdateNoteData {
  title?: string
  content?: string
  note_type?: string
  status?: string
}

export interface MiaojiParseResult {
  characters: Array<{
    name: string
    role_type: string
    gender?: string
    age?: string
    occupation?: string
    personality_traits?: string
    appearance?: string
    background?: string
  }>
  worldbuilding: Array<{
    name: string
    category: string
    description?: string
  }>
  outline: Array<{
    title: string
    summary?: string
    sort_order?: number
  }>
  events: Array<{
    title: string
    description?: string
    event_type?: string
    importance?: number
  }>
  summary: string
}

// 获取笔记列表
export async function getNotes(projectId: number, noteType?: string): Promise<Note[]> {
  const params = noteType ? `?note_type=${noteType}` : ''
  return request.get<Note[]>(`/projects/${projectId}/notes/${params}`)
}

// 获取单个笔记
export async function getNote(projectId: number, noteId: number): Promise<Note> {
  return request.get<Note>(`/projects/${projectId}/notes/${noteId}/`)
}

// 创建笔记
export async function createNote(projectId: number, data: CreateNoteData): Promise<Note> {
  return request.post<Note>(`/projects/${projectId}/notes/`, data)
}

// 更新笔记
export async function updateNote(projectId: number, noteId: number, data: UpdateNoteData): Promise<Note> {
  return request.put<Note>(`/projects/${projectId}/notes/${noteId}/`, data)
}

// 删除笔记
export async function deleteNote(projectId: number, noteId: number): Promise<void> {
  return request.delete(`/projects/${projectId}/notes/${noteId}/`)
}

// 快速创建妙记
export async function quickMiaoji(projectId: number, content: string): Promise<Note> {
  return request.post<Note>(`/projects/${projectId}/notes/miaoji/quick`, { content })
}

// 解析妙记
export async function parseMiaoji(projectId: number, noteId: number): Promise<MiaojiParseResult> {
  return request.post<MiaojiParseResult>(`/projects/${projectId}/notes/${noteId}/parse/`)
}